### PR TITLE
[MNT] Update codeowners

### LINKED ISSUE #9: [FORK] Changing codeowners
I suggest we just delete the current version
https://github.com/scikit-time/scikit-time/blob/main/CODEOWNERS
and give admins ownership of everything. Any thoughts?

### PR DESCRIPTION
**Pull request by TonyBagnall**
Updates codeowners, closes #9 

Codeowners main use is to tag people on new PRs so things dont get overlooked. Moving towards at least one core dev per module. Need to resolve how recursive it is, confirm that if core dev A owns aeon/X they do not get tagged if someone else owns aeon/X/Y.

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -1,116 +1,68 @@
...
-* @aiwalter @guzalbulatova
-
+aeon/annotation @lmmentel
+aeon/annotation/clasp.py @patrickzib @ermshaua
 
-aeon/regression/dummy/ @badrmarani
-aeon/regression/feature_based/_fresh_prince.py @dguijo
-aeon/regression/deep_learning/inception_timep.py @hadifawaz1999
-aeon/regression/deep_learning/tests/test_deep_regressor_base.py @hadifawaz1999
+aeon/benchmarking @TonyBagnall @MatthewMiddlehurst @hadifawaz1999
```
Comment (MatthewMiddlehurst) [inline CODEOWNERS:7]: space here

File: CODEOWNERS [hunk 2]
```diff
@@ -1,116 +1,68 @@
...
-aeon/annotation/clasp.py @patrickzib @ermshaua
-aeon/annotation/igts.py @lmmentel
-aeon/annotation/ggs.py @lmmentel
-aeon/annotation/datagen.py @lmmentel
+aeon/transformations/collection @MatthewMiddlehurst @TonyBagnall
+aeon/transformations/collection/augmenter.py @MrPr3ntice @iljamaurer
+aeon/transformations/collection/dictionary_based/_paa.py @patrickzib @MatthewMiddlehurst
+aeon/transformations/collection/dictionary_based/_sax.py @patrickzib @MatthewMiddlehurst @hadifawaz1999
+aeon/transformations/collection/dictionary_based/_sfa.py @patrickzib @MatthewMiddlehurst
+aeon/transformations/collection/dictionary_based/_sfa_fast.py @patrickzib @MatthewMiddlehurst
```
Comment (MatthewMiddlehurst) [inline CODEOWNERS:40]: this can be the whole package probably?

File: CODEOWNERS [hunk 3]
```diff
@@ -1,116 +1,68 @@
...
-aeon/transformations/series/kalman_filter.py @NoaBenAmi
-aeon/transformations/panel/augmenter.py @MrPr3ntice @iljamaurer
-aeon/transformations/multiplex.py @miraep8
-aeon/transformations/tests/test_multiplexer.py @miraep8
-aeon/transformations/panel/channel_selection.py @haskarb @a-pasos-ruiz @TonyBagnall
-aeon/transformations/series/time_since.py @KishManani
-aeon/transformations/series/pca.py @aiwalter
-
-aeon/forecasting/base/ @mloning @aiwalter
+aeon/forecasting @aiwalter @guzalbulatova
```
Comment (MatthewMiddlehurst) [inline CODEOWNERS:17]: Maybe @ltsaprounis if he is willing. I wouldn't say this is an obligation to review every ping.

File: CODEOWNERS (hunk 2 above)
Comment (hadifawaz1999) [reply CODEOWNERS:40]: if you want yeah though i only contributed to PAA and SAX.

**Review by hadifawaz1999 — APPROVED** [review]
LGTM!

**Review by MatthewMiddlehurst — APPROVED** [review]
I'm fine with the changes. Not like they are permanent, anyone is welcome to remove themselves as any time.

**Pull request merged**